zebra-consensus/src/block/check.rs

```diff
@@ -115,18 +112,29 @@ pub fn subsidy_is_valid(block: &Block, network: Network) -> Result<(), BlockErro
         )
     } else if halving_div.count_ones() != 1 {
         unreachable!("invalid halving divisor: the halving divisor must be a non-zero power of two")
-    } else if height < canopy_activation_height {
+    } else if subsidy::founders_reward::founders_reward_active(height, network) {
         // Founders rewards are paid up to Canopy activation, on both mainnet and testnet
         let founders_reward = subsidy::founders_reward::founders_reward(height, network)
             .expect("invalid Amount: founders reward should be valid");
         let matching_values = subsidy::general::find_output_with_amount(coinbase, founders_reward);
 
-        // TODO: the exact founders reward value must be sent as a single output to the correct address
-        if !matching_values.is_empty() {
-            Ok(())
-        } else {
-            Err(SubsidyError::FoundersRewardNotFound)?
+        let founders_reward_address =
+            subsidy::founders_reward::founders_reward_address(height, network)
+                .expect("we should have an address");
+        let matching_address =
+            subsidy::founders_reward::find_output_with_address(coinbase, founders_reward_address);
+
+        if matching_values.is_empty() {
+            return Err(SubsidyError::FoundersRewardAmountNotFound)?;
+        }
+        if matching_address.is_empty() {
+            return Err(SubsidyError::FoundersRewardAddressNotFound)?;
+        }
+        if matching_values != matching_address {
+            return Err(SubsidyError::FoundersRewardDifferentOutput)?;
         }
```

Comment:
This needs to be a set intersection test, not an equality test.

It is ok to:
* send the correct value to a different address
* send other values to the correct address
* send multiple correct values to the correct address

As long as at least one correct value is sent to the correct address.

Please also add tests for all these cases.

As an alternative, we could checkpoint on Canopy, and delete this code. (But we'll probably want to re-use this code in the Funding Streams implementation, so we might as well fix it now.)